adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDDFunctions.scala

```diff
@@ -49,6 +51,65 @@ class NucleotideContigFragmentRDDFunctions(rdd: RDD[NucleotideContigFragment]) e
   }
 
   /**
+   * Save nucleotide contig fragments in FASTA format.
+   *
+   * @param fileName file name
+   */
+  def adamSaveAsFasta(fileName: String) = {
+
+    def isFragment(record: NucleotideContigFragment): Boolean = {
+      (Option(record.fragmentNumber), Option(record.numberOfFragmentsInContig)) match {
+        case (None, None)       => false
+        case (None, Some(m))    => false
+        case (Some(n), None)    => false
+        case (Some(n), Some(m)) => (m > 1)
+      }
+    }
+
+    def toFasta(record: NucleotideContigFragment): String = {
+      val sb = new StringBuilder()
+      sb.append(">")
+      sb.append(record.contig.contigName)
+      Option(record.description).foreach(n => sb.append(" ").append(n))
+      if (isFragment(record)) {
+        sb.append(" fragment %d of %d".format(record.fragmentNumber + 1, record.numberOfFragmentsInContig))
```

Comment:
If you just call ```saveAsFasta``` on an RDD it will write out separate FASTA records for each fragment.  E.g. https://github.com/bigdatagenomics/adam/pull/816/files#diff-26ce0c60df6e82a09e1c8f5e491e58b3R356

The adam2fasta command calls ```rdd.mergeFragments().saveAsFasta(fileName)```, or soon something like ```rdd.mergeFragments().coalesce(partitions, shuffle).saveAsFasta(fileName)```.  Suppose merge fragments could be a command-line argument as well.